addon/mixins/route.js

```diff
@@ -217,6 +216,19 @@ export default Ember.Mixin.create({
     return false;
   },
 
+  /**
+   Update the infinity model with new objects
+
+   @method updateInfinityModel
+   @param {Ember.Enumerable} newObjects The new objects to add to the model
+   @return {Ember.Array} returns the updated infinity model
+   */
+  updateInfinityModel(newObjects) {
+    var infinityModel = this.get(this.get('_modelPath'));
+
+    return infinityModel.pushObjects(newObjects);
+  },
+
```

Comment:
Ahhh now the problem is that we've pulled the `newObjects.get('content')` back into the load method.  That means, if someone is using pure arrays without prototype extensions, there won't be a good place for them to override the `get('content')` part of the push.

I like this idea because it exposes the method for actually extracting the data from the collection that is pushed into the `infinityModel`, giving developers a public place to change that logic.